api/nvidia.com/resource/gpu/v1alpha1/gpuconfig.go

```diff
@@ -45,11 +46,26 @@ func DefaultGpuConfig() *GpuConfig {
 				Interval: ptr.To(DefaultTimeSlice),
 			},
 		},
+		DriverConfig: &GpuDriverConfig{
+			Driver: NvidiaDriver,
+		},
 	}
 }
 
 // Normalize updates a GpuConfig config with implied default values based on other settings.
 func (c *GpuConfig) Normalize() error {
+	if c.DriverConfig == nil {
+		c.DriverConfig = DefaultGpuDriverConfig()
+	} else {
+		if err := c.DriverConfig.Normalize(); err != nil {
+			return err
+		}
+	}
+	// If driver is not Nvidia, don't proceed with normalizing sharing configuration.
+	if c.DriverConfig.Driver != NvidiaDriver {
+		return nil
+	}
```

Comment:
Also, is the expectation that we always call `Validate()` after `Normalize()` so that we check whether there is no sharing actually configured? Does it make sense to remove this check here entirely so that we only need to update logic in one place with regards to the driver type and sharing interaction?

Reply:
> Does it make sense to remove this check here entirely so that we only need to update logic in one place with regards to the driver type and sharing interaction?

Removing the check here won't work. There's 2 possibilities here:
1. On NVIDIA driver, normalize sharing by initializing it if its nil, and then set config based on strategy.
2. On VFIO-PCI driver, don't initialize it.

If we allow normalization of `c.Sharing` in case (2) by relaxing the check, then we're going to fail subsequent validation.